laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -469,43 +377,53 @@ public function updateMilestones($id){
             ->orderBy('last_name')
             ->get();
 
-        $datas = compact('data1', 'data2', 'data3');
+        $datas = $resident_data;
 
         return view('schedules.resident.milestone', compact('id', 'milestones', 'datas', 'anesthesiologists'));
-    }
+    } 
 
-	public function notifyResidentOverwrittenPreferences($toName, $toEmail, $residentName, $date, $overwrittenChoices)
+    public function notifyResidentOverwrittenPreferences($toName, $toEmail, $residentName, $date, $overwrittenChoices)
     {
 
-		$choice = "";
-		if($overwrittenChoices[0] == 1){
-			$choice = "1";
-		}
-
-		if($overwrittenChoices[1] == 2){
-			$choice = $choice." 2";
-		}
-
-		if($overwrittenChoices[2] == 3){
-			$choice = $choice." 3";
-		}
+        $choice = "";
+        //$choice = implode(" ", $overwrittenChoices);
+        for ($i = 1; $i <= 3; $i++){
+            if ($overwrittenChoices[$i - 1] != 0) {
+                $choice = $choice . $i . " ";
+            }
+        }
 
-		$subject = 'REMODEL: Resident Preference '.$choice.' Overwritten for '.$date;
-		$body = "Resident $residentName has overwritten OR preferences  ".$choice." for ".$date.". New preferences are now viewable on REMODEL website.";
-		$heading = "Resident $residentName has overwritten OR preference ".$choice;
-        $data = array('name'=>$toName, 'heading'=>$heading, 'body'=>$body);
+        $subject = 'REMODEL: Resident Preference ' . $choice . ' Overwritten for ' . $date;
+        $body = "Resident $residentName has overwritten OR preferences  " . $choice . "for " . $date . ". New preferences are now viewable on REMODEL website.";
+        $heading = "Resident $residentName has overwritten OR preference " . $choice;
+        $data = array('name' => $toName, 'heading' => $heading, 'body' => $body);
 
-        Mail::send('emails.mail', $data, function($message) use ($toName, $toEmail, $subject) {
+        Mail::send('emails.mail', $data, function ($message) use ($toName, $toEmail, $subject) {
             $message->to($toEmail, $toName)->subject($subject);
             $message->from('OhioStateAnesthesiology@gmail.com');
         });
-		return true;
+        return true;
     }
 
     // Insert options if no preference exists.
-    // Update options when overwriting preferences.
     private function insertOption()
     {
+
+        // variables to track if the use has overwritten a preference
+        $notify = false;
+        $overwrittenChoices = array(0,0,0);
```

Comment:
Maybe instead of having the array default be `array(0,0,0)`  have it be `array()` and then use `array_push($overwrittenChoices, $i);`  where `$i` would either 1,2,3 depending on choice. You wouldn't have an `array_push `in the else, it would only be in the if.
 
Then we can just use `implode(" ", $overwrittenChoices);`  in the `notifyResidentOverwrittenPreferences` method.